src/Utilities/Compiler/Extensions/ISymbolExtensions.cs

```diff
@@ -659,6 +659,20 @@ public static bool HasAnyAttribute(this ISymbol symbol, params INamedTypeSymbol?
             return symbol.GetAttributes(attributeTypesToMatch).Any();
         }
 
+        public static bool HasAnyAttribute(this ISymbol symbol, [NotNullWhen(true)] INamedTypeSymbol? attributeToMatch)
```

Comment:
What is the difference between this extension, and the one right below? https://github.com/dotnet/roslyn-analyzers/blob/1d9fd4e26f90c271b20560613808cac1745e7422/src/Utilities/Compiler/Extensions/ISymbolExtensions.cs#L662-L680

Reply:
➡️ I'll send a follow-up to address this.

Reply:
➡️ To answer the question, the previous `HasAttribute` method was delegating to the `params` form of `HasAnyAttribute`. The new method avoids the allocation that all calls to `HasAttribute` were facing.